pkg/ipmi/ipmi.go

```diff
@@ -173,3 +195,96 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) waitForSetInProgressCleared() error {
+	req := &req{}
+	req.msg.cmd = _GET_SYSTEM_INFO_PARAMETERS
+	req.msg.netfn = _IPMI_NETFN_APP
+	var data [4]byte
+
+	data[0] = 0
+	data[1] = _SET_IN_PROGRESS
+	data[2] = 0
+	data[3] = 0
+	req.msg.data = unsafe.Pointer(&data)
+	req.msg.dataLen = 4
+
+	retry := 20
+	for retry > 0 {
+		recv, err := i.sendrecv(req)
+		if err != nil {
+			return err
+		}
+		// response data byte 3 bit[1:0] == 00b indicates set complete
+		if len(recv) == 3 && (recv[2]&0x3) == 0 {
+			return nil
+		}
+		time.Sleep(1 * time.Millisecond)
+		retry--
+	}
+
+	return fmt.Errorf("Wait for Set in Progress cleared timeout")
+
+}
+
+func (i *IPMI) SetSystemFWVerToBmc() error {
+	strlen_max := 64 // Set 64 Bytes as the maximal version string length
+	entryData, tableData, err := getSmbiosData()
```

Comment:
In my opinion, this package should not depends on the `smbios` one. This function should be a generic Setsysinfo function and take a `[]byte` or string with the info you want to set.